### PR TITLE
ft-168781838 like/unlike an accommodation

### PR DESCRIPTION
**Pull request by alexismajya**
#### What does this PR do?

- As a user, I will be able to like/unlike an accommodation facility

#### Description of Task to be completed?

- create route `POST:api/v1/likings/:accommodationId

#### How should this be manually tested?
using `postman`: test =>
route: POST:api/v1/accommodations/:accommodationId/like

#### What are the relevant pivotal tracker stories?

[story](https://www.pivotaltracker.com/story/show/168781838)

#### Screenshots (if appropriate)
Screen Shot 2019-11-07 at 12 02 38

#### Questions:

### CODE REVIEW

File: src/database/migrations/20191104212522-create-liking-accommodation.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Likings',{
```
Comment (lemoissonM) [inline src/database/migrations/20191104212522-create-liking-accommodation.js:5]: great job Alexis. However, why don't you name this table likes?

File: src/routes/liking.route.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const app = express.Router();
+
+const { checkToken } = UserMiddleware;
+const { checkUserLiking } = LikingMiddleware;
+const { addLikeAccommdation } = LikingsController;
+
+app.post('/:accommodationId', checkToken, checkUserLiking, addLikeAccommdation);
+
+export default app;
```
Comment (lemoissonM) [inline src/routes/liking.route.js]: Why don't you put this into the accommodation route. Don't think it's important to create another file. I suggest to have a route like this : /accommodations/accommodationId/like or /accommodations/accommodationId/unlike

**Review by lemoissonM — CHANGES_REQUESTED** [review]

File: src/database/models/likingAccommodation.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  const Likings = sequelize.define('Likings', {
```
Comment (Crispy-rw) [inline src/database/models/likingAccommodation.js]: Great job @alexismajya  but liking is not a plural of like,
please change fix this

Comment (alexismajya) [reply src/database/models/likingAccommodation.js]: cool

Comment (alexismajya) [reply src/database/models/likingAccommodation.js]: Thank you Christian, for the feedback. however, according to our team naming conventions, the models should be singular

Comment (Crispy-rw) [reply src/database/models/likingAccommodation.js]: yes that's why it should be named `like`

File: src/routes/liking.route.js (hunk 2 above)
Comment (alexismajya) [reply src/routes/liking.route.js]: Thank you @lemoissonM  for your feedback. I already worked on that

**Review by lemoissonM — APPROVED** [review]

**Review by Crispy-rw — APPROVED** [review]

File: src/database/migrations/20191104212522-create-liking-accommodation.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      id: {
+        allowNull: false,
+        autoIncrement: true, 
+        primaryKey: true, 
+        type: Sequelize.INTEGER 
+      },
+      like: {
```
Comment (elchroy) [inline src/database/migrations/20191104212522-create-liking-accommodation.js]: Let this column name be more like a boolean. Use `isLiked` or `status`

File: src/database/migrations/20191104212522-create-liking-accommodation.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        type:Sequelize.BOOLEAN
+      },
+      userId: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        defaultValue: 0,
```
Comment (elchroy) [inline src/database/migrations/20191104212522-create-liking-accommodation.js]: Let the default value here be `1` or `true`

File: src/controllers/likings.controller.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    const { accommodationId } = req.params;
+    const { like } = req.params;
+    let likeValue;
+    if (like === 'like') {
+      likeValue = true;
+    } else if (like === 'unlike') {
+      likeValue = false;
+    } else {
+      return sendResult(res, 400, 'Invalid parameter');
+    }
```
Comment (elchroy) [inline src/controllers/likings.controller.js]: We don't need this. All we need is
- If the user has not liked it before, we create a new Like instance, with status as `true`
- If the user has liked it before, we toggle the like status
- i.e change from `true` to `false` or `false` to `true`.
- [findOrCreate](https://sequelize.org/master/manual/models-usage.html#-code-findorcreate--code----search-for-a-specific-element-or-create-it-if-not-available)

File: src/middlewares/liking.middleware.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import sendResult from '../utils/sendResult';
+import LikingService from '../services/liking.service';
+
+const LikingMiddleware = {
+  async checkUserLiking(req, res, next) {
```
Comment (elchroy) [inline src/middlewares/liking.middleware.js]: We don't really need a middleware here

**Pull request merged**